src/ir_Zepeal.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Christian Nilsson (nikize)
+
+/// @file
+/// @brief Support for Zepeal protocol.
+/// This protocol uses fixed length bit encoding.
+/// Company behinde this brand seems to be Denkyosha
+/// @see https://www.denkyosha.co.jp/
+
+// Supports:
+//   Brand: Zepeal,  Model: DRT-A3311(BG) floor fan
+//   Brand: Zepeal,  Model: DRT-A3311(BG) 5 button remote
+
+#include "IRrecv.h"
+#include "IRsend.h"
+#include "IRutils.h"
+
+// Constants
+
+const uint16_t kZepealHdrMark = 2330;
+const uint16_t kZepealHdrSpace = 3380;
+const uint16_t kZepealOneMark = 1300;
+const uint16_t kZepealZeroMark = 420;
+const uint16_t kZepealOneSpace = kZepealZeroMark;
+const uint16_t kZepealZeroSpace = kZepealOneMark;
+const uint16_t kZepealFooterMark = 420;
+const uint16_t kZepealGap = 6750;
+
+const uint8_t  kZepealTolerance = 40;
+
+const uint8_t kZepealCheckExpected = 0x6C;
+
+#if SEND_ZEPEAL
+/// Send a Zepeal formatted message.
+/// Status: STABLE / Works on real device.
+/// @param[in] data The message to be sent.
+/// @param[in] nbits The bit size of the message being sent.
+/// @param[in] repeat The number of times the message is to be repeated.
+void IRsend::sendZepeal(uint64_t data, uint16_t nbits, uint16_t repeat) {
+  sendGeneric(kZepealHdrMark, kZepealHdrSpace,
+              kZepealOneMark, kZepealOneSpace,
+              kZepealZeroMark, kZepealZeroSpace,
+              kZepealFooterMark, kZepealGap,
+              data, nbits, 38, true, repeat, kDutyDefault);
+}
+#endif  // SEND_ZEPEAL
+
+#if DECODE_ZEPEAL
+/// Decode the supplied Zepeal message.
+/// Status: STABLE / Works on real device.
+/// @param[in,out] results Ptr to the data to decode & where to store the decode
+///   result.
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect. Typically kZepealBits.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return A boolean. True if it can decode it, false if it can't.
+bool IRrecv::decodeZepeal(decode_results *results, uint16_t offset,
+                          const uint16_t nbits, const bool strict) {
+  if (results->rawlen <= 2 * nbits + kHeader - 1 + offset)
```

Comment:
It also has a footer, so add `+ kfooter`

Reply:
using `if (results->rawlen <= 2 * nbits + kHeader + kFooter - 1 + offset)`
The first real test fails
Correct is `if (results->rawlen < 2 * nbits + kHeader + kFooter - 1 + offset)` ?
Similar issues seems to exist in different protocols as well.

Reply:
Ack. Yeah `<` makes sense.